sanity-io/structured-content-2022
# Fallback OpenGraph image meta tags

# Fallback OpenGraph image meta tags

## Intent

- Fixes [this](https://app.shortcut.com/sanity-io/story/15577/handle-open-graph-image) Story.
- Creates `Figure` type and replaces all inline Figure declarations with references to the new type
- Outputs `og:image:width` and `og:image:height` meta tags for the image

## Description

In my browser, the image linked to in the meta tags for [/shared-sections-test][deploy-test] is 630x630, and we don't have the dimension data available, so we can't output `og:image:width`/-`height` tags for non-fallback images. There is [an issue](https://github.com/sanity-io/sanity/issues/1627) for this for the `createImageUrlBuilder`. ~Maybe omitting the tags is better?~ I did so in 72ad10a49ff2be8a0f569e37ed7585a6aa7f6916.

## Testing this PR

1. Visit the [frontpage][deploy]
2. Verify that there are three `og:image` meta tags, and that they have the correct values corresponding to the fallback image's attributes.
3. Visit [/shared-sections-test][deploy-test]
4. Verify that the meta tags hold the info of [the Route's](https://stagingadmin.structuredcontent.live/desk/route;36c8bda6-e3a3-4671-a260-cdaf3dcd2e43) associated image, not the fallback image's

[deploy]: https://structured-content-2022-web-j60fk11gg.sanity.build/
[deploy-test]: https://structured-content-2022-web-j60fk11gg.sanity.build/
